File: 프로그래머스/1/176963. 추억 점수/추억 점수.swift
import Foundation

func solution(_ name:[String], _ yearning:[Int], _ photo:[[String]]) -> [Int] {
    var rst: [Int] = []
    var dict: [String:Int] = [:]
    // init dict
    for i in 0..<name.count {
        dict[name[i]] = yearning[i]
    }
    
    for p in photo {
        var tempSum = 0
        for item in p {
            if dict[item] != nil {
                tempSum += dict[item]!
            }
        }
        rst.append(tempSum)
    }
    
    return rst
}

File: 프로그래머스/1/176963. 추억 점수/추억 점수.js
function solution(name, yearning, photo) {
    let answer = [];
    
    // :::: obj 구성 ::::
    // photo, yearning 기준
    let obj = {};
    for(let i=0; i < name.length; i++){
        obj[name[i]] = yearning[i];
    }
    // :::: test obj init ::::
    // console.log(obj);
    
    for(let row of photo) {
        let tmp = 0;
        // console.log(row);
        for(let el of row){
            if(obj[el]) {
         		tmp += obj[el];       
            }
        }
        // :::: add row sum to answer ::::
        answer.push(tmp);
    }
    
    return answer;
}